cypress/integration/query_builder_test.js

```diff
@@ -0,0 +1,190 @@
+    beforeEach(function() {
+      cy.visit("/biotestmine/querybuilder");
+    });
+
+    it("allows you to select data type and summary attributes with model browser", function() {
+      cy.get('.model-browser-root').within(() => {
+        cy.get('select').then($option => {$option.val("Protein")}).parent().trigger('change');
+        cy.contains("Summary").click();
+      });
+    });
+
+    it("can save and load query", function() {
+      cy.get('.model-browser-root').within(() => {
+        cy.get('select').then($option => {$option.val("Protein")}).parent().trigger('change');
+        cy.contains("Summary").click();
+      });
+
+      cy.get('div.panel-body').within(()=>{
+        cy.contains("Save Query").click();
+        cy.get("div.input-group").type("Protein summary", {delay:100});
+        cy.contains("Save Query").click();
+        cy.contains("Clear Query").click();
+      });
+        
+      cy.contains("Saved Queries").click();
+      cy.get('table').contains('td','Protein summary'); //Assertion
+      cy.contains('Load').click();
+    });
+
+    it("allows you to choose specific attributes", function(){
+      cy.get('.model-browser-root').within(() => {
+        cy.get('select').then($option => {$option.val("Gene")}).parent().trigger('change');
+        cy.get('select').should('have.value', 'Gene');
+        cy.contains("Summary").click();
+      });
+
+      cy.get('.model-browser').within(() => {
+        cy.get("ul").its('length').should("be.gt", 0);
+        cy.get(".qb-class").contains('UTRs').click();
+        cy.get('.expanded-group > ul').find('.qb-label').filter(':contains("Name")').click();
+        cy.get('.expanded-group > ul').find('.qb-label').filter(':contains("Length")').click();
+        cy.get('.expanded-group > ul').find('.qb-label').filter(':contains("DB identifier")').click();
+      });
+
+      cy.get('.query-view-column').within(() => {
+        cy.get('.qb-label').should('include.text','Name');
+        cy.get('.qb-label').should('include.text','Length');
+        cy.get('.qb-label').should('include.text','DB identifier');
+      });
+    });
+
+    it("allows selection of data type from data model", function(){
+      cy.contains("Data Model").click();
+      cy.contains("Gene").click();
+      cy.contains("Summary").click();
+
+      cy.get('.query-view-column').within(() => {
+        cy.get('.qb-label').should('include.text','Gene');
+      })
+    });
+
+    it('can add filters on query editor', function(){
+      cy.get('.model-browser-root').within(() => {
+        cy.get('select').then($option => {$option.val('Gene')}).parent().trigger('change');
+        cy.get('select').should('have.value', 'Gene');
+        cy.contains("Summary").click();
+      });
+
+      cy.get('.query-view-column').within(() => {
+        cy.get('.qb-label').should('include.text','Gene');
+        cy.get(':nth-child(1) > :nth-child(1) > :nth-child(1) > .lab > .icon-filter').click(); //rewrite
```

Comment:
Have you tried: `cy.contains('Organism').parents('.qb-label').siblings('.icon-filter').click()` where *Organism* is the attribute/class name?
Not much shorter, but at least the intent is clearer.

Reply:
I see now that you arrived at something similar a few lines below :smile: